client/src/client/rc_future.rs

```diff
@@ -65,8 +65,8 @@ impl<F> Clone for RcFuture<F>
 {
     fn clone(&self) -> Self {
         RcFuture {
-            rc_future: self.rc_future.clone(),
-            result: self.result.clone(),
+            rc_future: Rc::clone(&self.rc_future),
```

Comment:
this one doesn't look very clean to me... is there an explanation that goes with this one?

Reply:
The explanation for this one is: